src/main/java/org/reactome/release/dataexport/FTPFileUploader.java

```diff
@@ -0,0 +1,449 @@
+			deleteOldFilesFromServer();
+		} else {
+			getLogger().error("Unable to upload at least some files to the {} FTP server.  Old Files were not deleted",
+				getServerHostName());
+		}
+
+		logListingOfReactomeFilesPresentOnServer();
+		closeFTPConnectionToServer();
+	}
+
+	/**
+	 * Uploads the profile and links files (which match the pattern(s) defined by the method "isCurrentFile") for
+	 * Reactome data to the FTP Server.  Returns true if and only if all files are successfully uploaded; false
+	 * otherwise.
+	 *
+	 * @return <code>true</code> if all local files intended for the FTP Server are uploaded successfully,
+	 * <code>false</code> otherwise (including if some files are uploaded successfully, but at least one was not)
+	 * @throws IOException Thrown if unable to
+	 *  1) Get the names of the files from the local file system to upload
+	 *  2) Create an input stream to a local file to upload
+	 *  3) Store/upload the file on the FTP Server
+	 *  4) Get the current status from the FTP Server if attempting to upload a file fails without a thrown exception
+	 */
+	public boolean uploadFilesToServer() throws IOException {
+		List<String> filesToUpload = getLocalFileNamesToUpload();
+
+		if (filesToUpload.isEmpty()) {
+			getLogger().error("No files were found in the directory '{}' which should be uploaded to {}",
+				getLocalOutputDirectoryPath(), getServerHostName());
+
+			return false;
+		}
+
+		for (String fileToUpload : filesToUpload) {
+			if (!uploadFileToServer(fileToUpload)) {
+				return false; // File failed to upload - indicates not all files were uploaded successfully
+			};
+		}
+
+		return true; // All files uploaded successfully
+	}
+
+	/**
+	 * Deletes the profile and links files (which match the pattern(s) defined by the method "isPreviousFile" for
+	 * outdated Reactome data on the FTP Server.  Returns true if and only if all files are successfully deleted; false
+	 * otherwise.
+	 *
+	 * @return <code>true</code> if all old files intended to be deleted from the FTP Server are successfully deleted
+	 * or were found to already not exist on the server, <code>false</code> otherwise (including if some files are
+	 * successfully deleted, but at least one was not)
+	 * @throws IOException Thrown if unable to
+	 *  1) Get the names of the files from the remote FTP Server to delete
+	 *  2) Get the names of all Reactome specific files that currently exist on the remote FTP Server (before any files
+	 *  are attempted to be deleted)
+	 *  3) Delete a file on the FTP Server
+	 *  4) Get the current status message from the FTP Server if attempting to delete a file fails without a thrown
+	 *  exception
+	 */
+	public boolean deleteOldFilesFromServer() throws IOException {
+		List<String> filesToDelete = getRemoteFileNamesToDelete();
+		if (filesToDelete.isEmpty()) {
+			int previousReactomeVersion = getReactomeReleaseVersion() - 1;
+			getLogger().info("No files from Reactome version {} to delete on the FTP server {}",
```

Comment:
The logger object is declared in this class - why use the getter?

Reply:
Yeah, I will use the logger directly.  See https://github.com/reactome/release-data-exporter/pull/4#discussion_r515295006

Reply:
Done: https://github.com/reactome/release-data-exporter/pull/4/commits/23f8eba7e0885eeaa96062447cc5a181cedadf72